templates/CMakeLists.txt.template

```diff
@@ -164,7 +164,7 @@
   set(gRPC_INSTALL_INCLUDEDIR "include" CACHE STRING "Installation directory for headers")
   set(gRPC_INSTALL_CMAKEDIR "lib/cmake/<%text>${PACKAGE_NAME}</%text>" CACHE STRING "Installation directory for cmake config files")
   set(gRPC_INSTALL_SHAREDIR "share/grpc" CACHE STRING "Installation directory for root certificates")
-  set(gRPC_BUILD_MSVC_MP_COUNT 0 CACHE INTERNAL "The maximum number of processes for MSVC /MP option")
+  set(gRPC_BUILD_MSVC_MP_COUNT 0 CACHE STRING "The maximum number of processes for MSVC /MP option")
```

Comment:
note that the semantics of this option is a bit cryptic.
(0 = don't set, -1 "all processors available" and positive number = number of processes).

Also, have we investigated both `/MP<number_of_core>` and `/maxcpucount:<number_of_core>`?
(see https://github.com/scikit-build/scikit-build/issues/61)

Reply:
The `/maxcpucount` option is for the MSBUILD but cmake doesn't use it (it just calls `cl` directly) so it's irrelevant to us.

For the option, I'm aware that it's not that ideal but it's a minor option so I didn't spend much to make it clearer. Since it's documented, I'm okay with having it as it ;)